smac/epm/gp_kernels.py

```diff
@@ -27,11 +27,17 @@ def get_conditional_hyperparameters(X: np.ndarray, Y: Optional[np.ndarray]) -> n
 
 class MagicMixin:
 
-    prior = None
+    prior = None  # type: Optional[Prior]
 
-    def __call__(self, X, Y=None, eval_gradient=False, active=None):
+    def __call__(
+        self,
+        X: np.ndarray,
+        Y: Optional[np.ndarray] = None,
+        eval_gradient: bool = False,
+        active: Optional[np.ndarray] = None,
+    ) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
 
-        if active is None and self.has_conditions:
+        if active is None and self.has_conditions:  # type: ignore[attr-defined] # noqa F821
```

Comment:
That's because the mixin doesn't have the attribute `has_condition`. It is only there when the mixin is mixed in with a 'real' kernel. Shall I add a comment?

Reply:
Yes that would be helpful when reading through the code.